### ISSUE TITLE
Add support for executing all effects on other cards

### ISSUE THREAD

**Issue by micahstairs** [micahstairs/bga-innovation#100]
In the base set we only had effects which told the player to execute non-demand effects of other cards. In Artifacts, there are certain cards which also execute demand effects of other cards.

This change is currently blocked though, because we are waiting for clarity from Chris on how we decide who does the demand/compel effects on the other card. Which featured icon do we use? The one on the initially activated card or on the card being executed?

**Comment by micahstairs** [comment]
This work is also blocked by the database change in #45. For this reason, I will take this issue since making changes to the database is very delicate.

**Comment by micahstairs** [comment]
We got some clarity from Chris on this: ```Icon count for "who is eligible to share, be demanded of, be compelled" are locked in at the start of the Dogma action. Now that I'm thinking about it, yes, that's why it says "as if they were on this card" - they'll be shared/demanded/etc as if it were the featured icon.```

**Comment by micahstairs** [comment]
Yeager's Bell X-1A (age 9) is likely going to be the best card to implement as an example, since it has the extra complication that there's still more effect text after it executes another card. This might require extra columns in the nested execution table in order to remember which interaction we were on and how many were left.

**Comment by ultimatefiend** [comment]
Something isn't working with this...I can't execute all effects from stInterInteractionStep.  It doesn't seem to do anything.

**Comment by ultimatefiend** [comment]
The executeNonDemandEffects function doesn't appear to share properly.  The person that is sharing does it, but not the originator.

**Comment by micahstairs** [comment]
I found another bug with executeAllEffects when triggering a demand effect (getNextPlayerUnderEffect throws an error).

**Comment by micahstairs** [comment]
So I did some more testing on the original bug we found and discovered that with 3 players, only the first opponent shares the effect (and not the other opponent or the originator).

**Comment by micahstairs** [comment]
I think we are finally good here! If we discover any other bugs going forward, let's file new issues for them instead of re-opening this.
